### ISSUE TITLE
Feature request - Additional directories to search for sources

### ISSUE THREAD

**Issue by naemnamenmea** [Viatorus/quom#33]
Hi, I found this package very useful, but when trying to use it - encounter with a small problem. The case when sources are located in different directory then their headers - are not supported. The common folder structure of any project can be as follows:
_project/_
&nbsp;&nbsp;&nbsp;_public/_
&nbsp;&nbsp;&nbsp;_include/_
&nbsp;&nbsp;&nbsp;_src/_
So it would be nice if it will be possible to specify source directories relative to include directories. For instance via `--relative_source_dir` or `-SI` argument.
The most common use can be: `-SI "./" -SI "../" -SI "../src"`
By default the directory suffixes array is ["./'] but if at least one -SI argument provided it should be overwritten.

**Comment by Viatorus** [comment]
Thank you. I think this is possible.

I will look into this at the end of the week.

**Comment by naemnamenmea** [comment]
Very thank you for a quick feedback. Btw should it be adjustable via cli or some common suffixes hardcoded like yours extensions for source files - still space for thinking.

**Comment by Viatorus** [comment]
Sorry, I totally forgot this issue. I will do my best to address this asap (this weekend?).

**Comment by Viatorus** [comment]
Have a look at the first implementation.

**Comment by naemnamenmea** [comment]
Looks fine. Go ahead)

**Comment by Viatorus** [comment]
So with the new option _'--source_directory', '-S'_ there is now the possibility to either:

* set the source dir relative to the header file if you add a dot in front of the path like `./../src/`
* set the source dir relative to the working dir by not using a dot like `src`
* set the source dir absolute like `/home/dev/proj/src`

First relative to the header is checked (if set), then relative from working dir and absolute together (order of input)

Do you have a project of yours where this would work?

**Comment by naemnamenmea** [comment]
In my case all I need is the first option. I don`t think you need any concrete projects to test this feature. Some synthetic tests would be enough. Or you can pick up any project with consistent structure and without some complex logic of dependency inclusion.
I want to use this package in conjunction with cmake - to automatically substitute include directories of target dependencies.
In fact cmake have a facility to unity build but it doesn`t allow to obtain a single source file as an output (with all libs files a project depends on).

**Comment by Viatorus** [comment]
I just wanted to be sure to fulfill many project structure with this change. :)
I will create a new version this evening (GMT+2).

**Comment by Viatorus** [comment]
Version 1.3.1 is available. :)

**Comment by naemnamenmea** [comment]
I`ve tried but unsuccessfully :(
[quom-test-proj.zip](https://github.com/Viatorus/quom/files/6388072/quom-test-proj.zip)

**Comment by Viatorus** [comment]
Ah... are you working on windows?

**Comment by Viatorus** [comment]
This is my output (running on linux):

```cpp


// static lib headers

namespace staticLibPublicFile
{
    void f();
}

// dynamic lib headers

#define LibDynamicAPI __declspec(dllexport)

namespace dynamicLibPublicFile
{
    LibDynamicAPI void f();
}

#include <functional>
#include <iostream>
#include <tuple>

using namespace std;

int main()
{
#ifdef LOCAL_RUN
  cout << "Local run" << endl;
#else
  cout << "Remote run" << endl;
#endif

  {
    //auto [first, second] = []() -> pair<int, double> {return {};}();

    [out = ref(cout << "Hello ")] (){out.get() << "World!\n"; }();
  }

  // static lib
  {
    staticLibPublicFile::f();
  }

  // dynamic lib
  {
    dynamicLibPublicFile::f();
  }

  return 0;
}

namespace staticLibPrivateFile
{
    void f();
}

namespace staticLibPublicFile
{
    void f(){
        staticLibPrivateFile::f();
    }
}

namespace dynamicLibPrivateFile
{
    void f();
}

namespace dynamicLibPublicFile
{
    void f(){
        dynamicLibPrivateFile::f();
    }
}
#include <iostream>

namespace staticLibPrivateFile
{
    void f(){
        std::cout << "Hello from " << __FILE__ << std::endl;
    }
}
#include <iostream>

namespace dynamicLibPrivateFile
{
    void f(){
        std::cout << "Hello from " << __FILE__ << std::endl;
    }
}
```

**Comment by naemnamenmea** [comment]
Yes, i ran it on windows. And i have no code after main() function.

**Comment by Viatorus** [comment]
I could reproduce the issue on windows. Thank you. I will look into.

**Comment by Viatorus** [comment]
Should be fixed in 1.3.2

**Comment by naemnamenmea** [comment]
Ty, I confirm that it works on the provided example.
